packages/domain-picker/src/domain-picker/domain-i-own-item.tsx

```diff
@@ -0,0 +1,63 @@
+/**
+ * External dependencies
+ */
+import * as React from 'react';
+import { __ } from '@wordpress/i18n';
+import { Icon, chevronRight } from '@wordpress/icons';
+import { createInterpolateElement } from '@wordpress/element';
+import { useLocalizeUrl } from '@automattic/i18n-utils';
+
+/**
+ * Internal dependencies
+ */
+
+interface Props {
+	siteSlug: string;
+	source: string;
+}
+
+const DomainIOwn: React.FunctionComponent< Props > = ( { siteSlug, source } ) => {
```

Comment:
I know there is no linting rule for this but can the Component name [match the filename](https://airbnb.io/javascript/react/#naming)? Of course, the filename should remain in hyphen-case as per our convention.
Also, speaking of naming, `UseYourDomain` wouldn't be more relevant since it matches the name of the flow this component is linking to, at least for now?